query/analyze/annotations.feature

```diff
@@ -0,0 +1,458 @@
+          [Trunk({ $x: thing([subdummy]) })]
+        ]),
+        Not([Trunk({ $n: thing([name]), $x: thing([person]) })])
+      ])
+    ])
+    """
+
+
+  Scenario: Analyze returns the annotations of every stage in the query
+    Given connection open read transaction for database: typedb
+    When get answers of typeql analyze query
+      """
+      match
+        $x isa person, has name $n, has ref $r;
+        { $r == 2; } or { $r == 3; };
+      select $n, $x;
+      delete $n;
+      insert $x has name "John";
+      match $n1 isa name == "J";
+      put $x has name $n1;
+      """
+    Then analyzed query pipeline annotations are:
+    """
+    Pipeline([
+      Match([
+        Trunk({ $n: thing([name]), $r: thing([ref]), $x: thing([person]) }),
+        Or([
+          [Trunk({ $r: thing([ref]) })],
+          [Trunk({ $r: thing([ref]) })]
+        ])
+      ]),
+      Select(),
+      Delete([
+        Trunk({ $n: thing([name]), $x: thing([person]) })
+      ]),
+      Insert([
+        Trunk({ $_: thing([name]), $x: thing([person]) })
+      ]),
+      Match([
+        Trunk({ $n1:thing([name]), $x: thing([person]) })
+      ]),
+      Put([
+        Trunk({ $n1:thing([name]), $x: thing([person]) })
+      ])
+    ])
+    """
+
+
+  Scenario: Analyze returns the annotations of functions in the preamble
+    Given connection open read transaction for database: typedb
+    When get answers of typeql analyze query
+      """
+      with
+      fun names_of($p: person) -> { name }:
+      match $p has name $n;
+      return { $n };
+
+      match
+        $p isa person;
+        let $n in names_of($p);
+      """
+
+    Then analyzed preamble annotations contains:
+      """
+      Function(
+        [thing([person])],
+        stream([thing([name])]),
+        Pipeline([
+          Match([
+            Trunk({ $n: thing([name]), $p: thing([person]) })
+          ])
+        ])
+      )
+      """
+
+    Then analyzed query pipeline annotations are:
+      """
+      Pipeline([
+        Match([
+          Trunk({ $n: thing([name]), $p: thing([person]) })
+        ])
+      ])
+      """
+
+
+  Scenario: Analyze returns the annotations of fetch
+    # Basic concept
+    Given connection open read transaction for database: typedb
+    When get answers of typeql analyze query
+      """
+      match $n isa name;
+      fetch { "names": $n };
+      """
+    Then analyzed fetch annotations are:
+    """
+    { names: [string] }
+    """
+
+    # Basic value
+    When get answers of typeql analyze query
+      """
+      match let $x = 5;
+      fetch { "x": $x };
+      """
+    Then analyzed fetch annotations are:
+    """
+    { x: [integer] }
+    """
+
+    # Wildcard
+    When get answers of typeql analyze query
+      """
+      match $x isa person;
+      fetch {
+        $x.*
+      };
+      """
+    Then analyzed fetch annotations are:
+    """
+    {
+        name: [string],
+        ref: [integer]
```

Comment:
Im guessing these are all representing optionals right? Eg could be name-string or missing